app/src/main/java/com/example/rickandmorty/fragment/ListFragment.kt

```diff
@@ -0,0 +1,153 @@
+package com.example.rickandmorty.fragment
+
+import android.app.AlertDialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.rickandmorty.R
+import com.example.rickandmorty.adapter.CharacterAdapter
+import com.example.rickandmorty.addHorizontalSpaceDecoration
+import com.example.rickandmorty.addPaginationScrollListener
+import com.example.rickandmorty.databinding.FragmentListBinding
+import com.example.rickandmorty.model.CharacterList
+import com.example.rickandmorty.model.PagingData
+import com.example.rickandmorty.retrofit.RickAndMortyService
+import com.google.android.material.snackbar.Snackbar
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class ListFragment : Fragment() {
+    private var _binding: FragmentListBinding? = null
+    private val binding get() = requireNotNull(_binding) {
+        "View was destroyed"
+    }
+
+    private val adapter = CharacterAdapter { character ->
+        findNavController().navigate(
+            ListFragmentDirections.toDetails(character.id)
+        )
+    }
+
+    private var currentRequest: Call<CharacterList>? = null
+    private var isLoading = false
+    private var currentPage = 1
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentListBinding.inflate(inflater, container, false)
+            .also { binding ->
+                _binding = binding
+            }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        loadCharacters()
+
+        with(binding) {
+            toolbar.menu
+                .findItem(R.id.info_button)
+                .setOnMenuItemClickListener {
+                    showDialog()
+                    true
+                }
```

Comment:
лучше повесить обработку всего меню на тулбар
```suggestion
            toolbar.setOnMenuItemClickListener {
                if (it.itemId == R.id.info_button) {
                    showDialog()
                    true
                } else {
                    false
                }
            }
```